### PR TITLE
Added sign up logic and changed to use MySQL DB

### PR DESCRIPTION
**Pull request by bananapizzza**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   "dependencies": {
     "cors": "^2.8.5",
     "express": "^4.17.1",
-    "jquery": "^3.4.1"
+    "jquery": "^3.4.1",
```
Comment (xaveng) [inline package.json]: Why do we need jquery?

Comment (xaveng) [reply package.json]: npm uninstall jquery

File: src/backendLogic.js [hunk 2]
```diff
@@ -5,23 +5,30 @@ const path = require('path');
...
+    host: "localhost",
+    user: "bananapizzza",
+    password: "1234",
+    database: "random_workout"
+});
+
 const readFileAsync = util.promisify(fs.readFile);
+const writeFileAsync = util.promisify(fs.writeFile);
 const rootPath = path.parse(appRoot).dir;
+ const queryDBAsync = util.promisify(pool.query).bind(pool);
```
Comment (xaveng) [inline src/backendLogic.js]: minor: indentation

File: src/backendLogic.js [hunk 3]
```diff
@@ -5,23 +5,30 @@ const path = require('path');
...
+const ALREADY_USED_ID = 'ALREADY_USED_ID';
+
+const pool = mysql.createPool({
+    host: "localhost",
+    user: "bananapizzza",
+    password: "1234",
+    database: "random_workout"
+});
+
 const readFileAsync = util.promisify(fs.readFile);
```
Comment (xaveng) [inline src/backendLogic.js:26]: Do we need this ?

File: src/backendLogic.js [hunk 4]
```diff
@@ -38,26 +45,102 @@ app.get('/', async (req, res) => {
...
-        res.send(JSON.stringify("Invalid User"));
+    const username = req.body.username;
+    const password = req.body.password;
+
+    switch (await checkLoginInfo(username, password)) {
+        case LOGIN_SUCCESS:
+            res.send(JSON.stringify(LOGIN_SUCCESS));
+            break;
+        case WRONG_PASSWORD:
+            res.statusCode = 401;
```
Comment (xaveng) [inline src/backendLogic.js]: Let's use this with http code name, not just numbers.
https://www.npmjs.com/package/http-status-codes

File: src/backendLogic.js [hunk 5]
```diff
@@ -38,26 +45,102 @@ app.get('/', async (req, res) => {
...
+
+async function checkSignUpInfo(username) {
+    return await (async () => {
+        try{
+            const sql = `SELECT * from users WHERE username=?`;
+            const queryResult = await queryDBAsync(sql, username);
+            if (queryResult.length > 0) {
+                return ALREADY_USED_ID;
+            } else {
+                return SIGN_UP_SUCCESS;
```
Comment (xaveng) [inline src/backendLogic.js]: Looks like we are checking if this user is in the DB or not. Let's make a separate keyword for this.

File: src/backendLogic.js [hunk 6]
```diff
@@ -38,26 +45,102 @@ app.get('/', async (req, res) => {
...
+
+async function addNewUserToDB(username, password) {
+    return await (async () => {
+        try {
+            const values = [username, password];
+            const sql = `INSERT INTO users(username, password) VALUES (?, ?)`;
+            await queryDBAsync(sql, values);
+            console.log("Added new user to DB");
+        } catch (err) {
+            console.log("addNewUserToDB err: "+err);
```
Comment (xaveng) [inline src/backendLogic.js]: minor: indentation.

**Review by xaveng — COMMENTED** [review]
As this backendLogic is getting larger, we could think of separating this file into each files. E.g) loginProcessor.js something like this. :)

File: package.json (hunk 1 above)
Comment (bananapizzza) [reply package.json]: done

File: src/backendLogic.js (hunk 3 above)
Comment (bananapizzza) [reply src/backendLogic.js:26]: deleted

File: src/backendLogic.js (hunk 5 above)
Comment (bananapizzza) [reply src/backendLogic.js]: done

File: src/backendLogic.js (hunk 4 above)
Comment (bananapizzza) [reply src/backendLogic.js]: okie

**Review by xaveng — APPROVED** [review]

**Pull request merged**